nessie/src/main/java/org/apache/iceberg/nessie/TableReference.java

```diff
@@ -83,12 +95,49 @@ public static TableReference parse(String path) {
     }
 
     if (path.contains("#")) {
-      throw new IllegalArgumentException("Invalid table name:" +
-          " # is not allowed (reference by timestamp is not supported)");
+      String[] tableTimestamp = path.split("#");
+      TableIdentifier identifier = TableIdentifier.parse(tableTimestamp[0]);
+      return new TableReference(identifier, parseTimestamp(tableTimestamp[1]), null);
     }
 
     TableIdentifier identifier = TableIdentifier.parse(path);
 
     return new TableReference(identifier, null, null);
   }
+
+  private enum FormatOptions {
+    DATE_TIME(DateTimeFormatter.ISO_DATE_TIME, Instant::from),
+    LOCAL_DATE_TIME(DateTimeFormatter.ISO_LOCAL_DATE_TIME, t -> LocalDateTime.from(t).atZone(UTC).toInstant()),
+    LOCAL_DATE(DateTimeFormatter.ISO_LOCAL_DATE, t -> LocalDate.from(t).atStartOfDay(UTC).toInstant());
```

Comment:
This doesn't seem specific enough to include. I would expect the ref on some day to be the ref at the end of that day, but I think this would produce the ref at the start of the day.

Reply:
hmmmm...I am torn. On one hand I agree, it makes the users life easier if we add 23:59:59 to the timestamp they specified. However I don't like changing what they asked for, if they ask for a date I think they should be given what they asked for, they can always specify time if they want EOD. Or they can create a tag to represent EOD. Thoughts?

Reply:
I'm fine changing this to 23:59:59, but there is also a good argument that it is ambiguous and should be left out. Up to you guys whether you want to do that or not. Sounds like @jacques-n is in favor of the prefix matching behavior.

Reply:
I think prefix matching is the better way to think about it. The datetime arithmetic is a bit fidgety now but I am happy with the test coverage.